### ISSUE TITLE
Saving in database fails almost every time

### ISSUE THREAD

**Issue by da-kami** [itchysats/itchysats#1218]
Rebased onto current master on `fund-fund-fund` branch.

Latest commits:

```
8e78fb87a92b4cac505e8e92ead4fd5dd05ba11e
53e87fdf71ff99cdacc50be49c120b58b8f93778
5eac778e197d02ca565250b9268309ae9c9fa56d (Avoid concurrent connections to database)
```

What I did:

* Start maker and taker daemon locally
* Create an offer in maker
* Go long in taker
* When hitting `Confirm` nothing happens because we seem to fail saving into the database

I see these errors in the maker logs:

```
2022-01-26 02:18:52 ERROR daemon::cfd_actors: Sending event to process manager failed: pool timed out while waiting for an open connection

...

2022-01-26 02:11:53  WARN daemon::projection: Failed to acquire DB connection: pool timed out while waiting for an open connection
2022-01-26 02:11:53 TRACE daemon::maker_inc_connections: Failed to send heartbeat because connection is gone taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442
2022-01-26 02:11:53 ERROR daemon::maker_inc_connections: Failed to read incoming messages from taker taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442
2022-01-26 02:11:55 TRACE daemon::monitor: Updating status of 2 transactions

...

2022-01-26 02:12:31 TRACE daemon::bitmex_price_feed: Received pong
2022-01-26 02:12:33  WARN daemon::projection: Failed to acquire DB connection: pool timed out while waiting for an open connection
2022-01-26 02:12:33  WARN daemon::maker_inc_connections: Refusing to accept 2nd connection from already connected taker ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442!
2022-01-26 02:12:33  WARN daemon::maker_inc_connections: Refusing to accept 2nd connection from already connected taker ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442!
2022-01-26 02:12:33  WARN daemon::maker_inc_connections: Refusing to accept 2nd connection from already connected taker ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442!
2022-01-26 02:12:33 ERROR daemon::maker_inc_connections: Failed to read incoming messages from taker taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442
2022-01-26 02:12:33  INFO daemon::maker_inc_connections: Dropping connection taker_id=ba4a7001bb1debc4e355f971e7e87409281865691ae64cbd9c8cfac1a07da442
2022-01-26 02:12:35 TRACE daemon::monitor: Updating status of 2 transactions

...

2022-01-26 02:18:52 ERROR daemon::cfd_actors: Sending event to process manager failed: pool timed out while waiting for an open connection
2022-01-26 02:18:57 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
```

Taker log errors:

```
...

2022-01-26 03:11:57 TRACE daemon::bitmex_price_feed: Received pong
2022-01-26 03:11:59 ERROR daemon::auto_rollover: Auto-rollover failed: pool timed out while waiting for an open connection
2022-01-26 03:12:02 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging

...


2022-01-26 03:11:57 TRACE daemon::bitmex_price_feed: Received pong
2022-01-26 03:11:59 ERROR daemon::auto_rollover: Auto-rollover failed: pool timed out while waiting for an open connection
2022-01-26 03:12:02 TRACE daemon::bitmex_price_feed: No message from BitMex in the last 5 seconds, pinging
2022-01-26 03:12:03 TRACE daemon::bitmex_price_feed: Received pong
...

```

I tried to open a position several times and keep running into the `pool connection timeout`. Unable to open a position at all.

Eventually the tile appears (after 10~ seconds) but the maker is never notified.

Given the error message I think this is related to saving the `Cfd` in the database.

**Comment by da-kami** [comment]
@thomaseizinger did you test https://github.com/itchysats/itchysats/pull/1211 by opening a CFD locally?

**Comment by da-kami** [comment]
Could reproduce this on master as well.

**Comment by thomaseizinger** [comment]
> @thomaseizinger did you test https://github.com/itchysats/itchysats/pull/1211 by opening a CFD locally?

I didn't. Surprising that the tests didn't catch that!

**Comment by da-kami** [comment]
> > @thomaseizinger did you test #1211 by opening a CFD locally?
> 
> I didn't. Surprising that the tests didn't catch that!

Yeah, I was also surprised. Maybe because for the tests we are using an in-memory db and don't use a file based db.

**Comment by klochowicz** [comment]
Fixed with #1226
